source/common/http/conn_manager_utility.cc

```diff
@@ -92,8 +92,30 @@ ConnectionManagerUtility::MutateRequestHeadersResult ConnectionManagerUtility::m
   if (!Utility::isUpgrade(request_headers)) {
     request_headers.removeConnection();
     request_headers.removeUpgrade();
+
     if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.sanitize_te")) {
-      request_headers.removeTE();
+      auto teHeader = request_headers.getTEValue();
+
+      if (!teHeader.empty()) {
+        auto hasTrailersTE = false;
+
+        auto teValues = absl::StrSplit(, ",");
+        for (const auto& teValue : teValues) {
+          auto parts = absl::StrSplit(teValue, ";"); // Handles cases like "chunked, trailers;q=0.5"
```

Comment:
Since this block of code only looks for "trailers" which does not have any parts, do we actually need to do this additional check?

Reply:
I think it's possible to add weight to `trailers` but I'm not sure?

Reply:
I don't thinks so, according [to the spec](https://datatracker.ietf.org/doc/html/rfc9110#field.te):

```
  TE                 = #t-codings
  t-codings          = "trailers" / ( transfer-coding [ weight ] )
  transfer-coding    = token *( OWS ";" OWS transfer-parameter )
  transfer-parameter = token BWS "=" BWS ( token / quoted-string )
```

It's either the literal value "trailers", or else it's a transfer-coding and a weight.

Reply:
Ok perfect! That simplifies the parsing. We should be good now